rust/src/ssh/ssh.rs

```diff
@@ -24,15 +24,36 @@ use crate::frames::Frame;
 use nom7::Err;
 use suricata_sys::sys::AppProto;
 use std::ffi::CString;
-use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::atomic::{AtomicBool, Ordering, AtomicU8};
+
+#[repr(C)]
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+#[allow(non_camel_case_types)]
+pub enum SshEncryptionHandling {
+    SSH_HANDLE_ENCRYPTION_DEFAULT = 0, // Disable raw content, continue tracking
+    SSH_HANDLE_ENCRYPTION_BYPASS = 1,  // Skip processing of flow, bypass if possible
+    SSH_HANDLE_ENCRYPTION_FULL = 2,    // Handle fully like any other protocol
+}
 
 static mut ALPROTO_SSH: AppProto = ALPROTO_UNKNOWN;
 static HASSH_ENABLED: AtomicBool = AtomicBool::new(false);
 
+static ENCRYPTION_BYPASS_ENABLED: AtomicU8 = AtomicU8::new(SshEncryptionHandling::SSH_HANDLE_ENCRYPTION_DEFAULT as u8);
```

Comment:
cc @jasonish is this the good rust way ?

Reply:
Yes, and no. Its the recommended way for a global static mutable, just making it a static mutable results in a compiler warning:
- See: https://github.com/OISF/suricata/pull/12673
- And: https://doc.rust-lang.org/nightly/edition-guide/rust-2024/static-mut-references.html

But perhaps not the best Suricata way if only modified on initialization, see:
https://github.com/OISF/suricata/pull/12665

Reply:
https://doc.rust-lang.org/nightly/edition-guide/rust-2024/static-mut-references.html#no_std-one-time-initialization looks pretty similar to what we do now, except adding a wrapper function to return a `const` (instead of mut)

Reply:
Still uses atomics right?

Reply:
> Still uses atomics right?

Not really, atomics are here not used for `static mut STATE`

They just want to add checks for the initialization of the global, but in our cases, it is done right I think.

Reply:
Yeah, seems as simple as this: https://github.com/OISF/suricata/pull/12680/commits/b486d55ba3cd015b40657829733d102586ead35d

Reply:
Looks good

Reply:
I was trying to understand what I should do but with all the references I got confused. Could you please be more direct? Thanks
My current understanding is that it is good as is, is that correct?

Reply:
I think you should use `static mut`